src/mcp/shared/session.py

```diff
@@ -455,6 +455,27 @@ async def _receive_loop(self) -> None:
                         pass
                 self._response_streams.clear()
 
+    def _normalize_request_id(self, response_id: RequestId) -> RequestId:
+        """
+        Normalize a response ID to match how request IDs are stored.
+
+        Since the client always sends integer IDs, we normalize string IDs
+        to integers when possible. This matches the TypeScript SDK approach:
+        https://github.com/modelcontextprotocol/typescript-sdk/blob/a606fb17909ea454e83aab14c73f14ea45c04448/src/shared/protocol.ts#L861
+
+        Args:
+            response_id: The response ID from the incoming message.
+
+        Returns:
+            The normalized ID (int if possible, otherwise original value).
+        """
+        if isinstance(response_id, str):
+            try:
+                return int(response_id)
+            except ValueError:
+                pass
```

Comment:
Should we log a warning or the exception here? If we hit this it means the assumption of integer `RequestId`s is violated right? Seems worth bubbling that up.

Reply:
Something like "response_id received is non-numeric: {value}"